Sts1CobcSw/Firmware/EduProgramTransferThread.cpp

```diff
@@ -0,0 +1,159 @@
+#include <Sts1CobcSw/Edu/Edu.hpp>
+#include <Sts1CobcSw/Edu/Types.hpp>
+#include <Sts1CobcSw/FileSystem/DirectoryIterator.hpp>
+#include <Sts1CobcSw/FileSystem/FileSystem.hpp>
+#include <Sts1CobcSw/Firmware/EduPowerManagementThread.hpp>
+#include <Sts1CobcSw/Firmware/StartupAndSpiSupervisorThread.hpp>
+#include <Sts1CobcSw/Firmware/ThreadPriorities.hpp>
+#include <Sts1CobcSw/Firmware/TopicsAndSubscribers.hpp>
+#include <Sts1CobcSw/FramSections/FramLayout.hpp>
+#include <Sts1CobcSw/FramSections/PersistentVariables.hpp>
+#include <Sts1CobcSw/Outcome/Outcome.hpp>
+#include <Sts1CobcSw/RodosTime/RodosTime.hpp>
+#include <Sts1CobcSw/Utility/DebugPrint.hpp>
+#include <Sts1CobcSw/Vocabulary/ProgramId.hpp>
+#include <Sts1CobcSw/Vocabulary/Time.hpp>
+
+#include <strong_type/difference.hpp>
+#include <strong_type/type.hpp>
+
+#include <rodos_no_using_namespace.h>
+
+#include <charconv>
+#include <cstdint>
+#include <system_error>
+#include <utility>
+
+
+namespace sts1cobcsw
+{
+namespace
+{
+constexpr auto stackSize = 1500U;
+constexpr auto eduProgramTransferThreadStartDelay = 15 * s;
+constexpr auto checkForProgramsInterval = 5 * s;
+
+
+[[nodiscard]] auto ReadProgramId(fs::DirectoryInfo const & info) -> Result<ProgramId>;
+auto HandleError(ErrorCode error) -> void;
+auto RemoveProgram(fs::Path const & file) -> void;
+auto SendProgramsToEdu(fs::DirectoryIterator & fileIterator) -> void;
+
+
+class EduProgramTransferThread : public RODOS::StaticThread<stackSize>
+{
+public:
+    EduProgramTransferThread()
+        : StaticThread("EduProgramQueueThread", eduProgramTransferThreadPriority)
+    {}
+
+
+private:
+    void run() override
+    {
+        SuspendFor(totalStartupTestTimeout);  // Wait for the startup tests to complete
+        SuspendFor(eduProgramTransferThreadStartDelay);
+        DEBUG_PRINT("Starting EDU program transfer thread\n");
+        TIME_LOOP(0, value_of(checkForProgramsInterval))
+        {
+            auto eduIsAlive = false;
+            eduIsAliveBufferForProgramTransfer.get(eduIsAlive);
+            if(edu::ProgramsAreAvailableOnCobc() && eduIsAlive)
+            {
+                DEBUG_PRINT("Programs available for transfer to EDU\n");
+                auto makeIteratorResult = fs::MakeIterator(edu::programsDirectory);
+                if(makeIteratorResult.has_error())
+                {
+                    DEBUG_PRINT("Failed to read program directory with error: %s\n",
```

Comment:
We write our debug messages in the form "<stuff that went wrong/failed>: <error code>", so please get rid of the "with error".